server/src/main/java/org/elasticsearch/search/aggregations/support/CoreValuesSourceType.java

```diff
@@ -390,5 +390,63 @@ public String typeName() {
     }
 
     /** List containing all members of the enumeration. */
-    public static List<ValuesSourceType> ALL_CORE = Arrays.asList(CoreValuesSourceType.values());
+    public static List<ValuesSourceType> ALL_CORE = Arrays.asList(values());
+
+    public enum ValueType implements Writeable {
```

Comment:
This look like it's just duplicated from ValueType?  I don't think we need a new, second, ValueType concept here.